### PR TITLE
Fix to sound not working in modern environments.

### PR DESCRIPTION
**Pull request by hasseily**
Need to use FAILED() instead of boolean check.

### CODE REVIEW

File: source/Mockingboard.cpp [hunk 1]
```diff
@@ -1248,10 +1248,11 @@ static void MB_UpdateInt(void)
 	DWORD dwDSLockedBufferSize0, dwDSLockedBufferSize1;
 	SHORT *pDSLockedBuffer0, *pDSLockedBuffer1;
 
-	if(!DSGetLock(MockingboardVoice.lpDSBvoice,
-						dwByteOffset, (DWORD)nNumSamples*sizeof(short)*g_nMB_NumChannels,
-						&pDSLockedBuffer0, &dwDSLockedBufferSize0,
-						&pDSLockedBuffer1, &dwDSLockedBufferSize1))
```
[the remark this answers is not in the document]
Comment (hasseily) [reply source/Mockingboard.cpp:1254]: Gaaah! My bad. I'd forgotten to push the fix to DSGetLock. Sorry about that.

**Comment by tomcw** [comment]
FYI, I'm currently doing some fairly involved refactoring of the Mockingboard speech support, so won't look at merging this PR until after I've gotten this speech refactoring merged into master. (ETA = soon!)

**Comment by tomcw** [comment]
@hasseily - the change I made in master recently (96bbc0c435af8242acd487606100a4b3b19ec500) split out the SSI263 code into a separate class. But this change resulted in a big merge conflict with your PR.

Options:
a) Do you want to redo your PR? (eg, drop this one, and create a new one)
b) Or I can manually make the changes you have proposed in this PR (but it'll mean that you won't be shown as the committer/contributor in the commit to master - but I will reference you in the commit comment).

Let me know what you want to do.

**Comment by hasseily** [comment]
Hi Tom, sorry this slipped through the cracks. Is this still of actuality? I'll make the changes

**Comment by hasseily** [comment]
Replaced with https://github.com/AppleWin/AppleWin/pull/951

**Pull request closed without merging**